### PR TITLE
Sm/cli supported metadata

### PR DESCRIPTION
**Pull request by mshanemc**
### What does this PR do?
this, mostly: https://github.com/salesforcecli/cli-supported-metadata
- a visual representation of our metadata support (SDR registry)
- vs the `test1` metadata coverage api (slightly different from the more well-used https://developer.salesforce.com/docs/metadata-coverage
- in the context of CLI/VSCode usage
new feature: forward looking types (current release + 1)
updating these files are part of SDR's build process (every push to github should rebuild the coverage files).  Thanks, husky!

### concerns I'd like feedback on
- that pre-push hook mucking with github release (post npm release push)
- if no active development is happening to SDR for a very long time, the files could get stale (registry types are correct but the comparison to the coverage reports is not)

### What issues does this PR fix or reference?
@W-10226996@

### CODE REVIEW

**Comment by WillieRuemmele** [comment]
we could change this to a weekly? cron job - that would resolve both of your concerns, right?

> - that pre-push hook mucking with github release (post npm release push)
> - if no active development is happening to SDR for a very long time, the files could get stale (registry types are correct but the comparison to the coverage reports is not)

**Comment by mshanemc** [comment]
> we could change this to a weekly? cron job - that would resolve both of your concerns, right?
> 
> > * that pre-push hook mucking with github release (post npm release push)
> > * if no active development is happening to SDR for a very long time, the files could get stale (registry types are correct but the comparison to the coverage reports is not)

definitely for the 2nd.  The downside to that would be it not running after SDR changes, which might be confusing for metadata owners.

Maybe it could run the job both weekly AND on SDR release (similar to or as part of the github release step).  Hmm.

**Review by shetzel — APPROVED** [review]
lgtm

File: scripts/update-registry/shared.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+export const getCurrentApiVersion = async () => {
+ return (
+    JSON.parse((await got(`https://mdcoverage.secure.force.com/services/apexrest/report`)).body) as {
+      versions: { selected: number };
+    }
+  ).versions.selected
+}
+
+export const getCoverage = async (apiVersion: number) => JSON.parse(
+    (await got(`https://na${apiVersion - 9}.test1.pc-rnd.salesforce.com/mdcoverage/api.jsp`)).body
```
[the remark this answers is not in the document]
Comment (mshanemc) [reply scripts/update-registry/shared.ts:13]: done.

**Review by WillieRuemmele — APPROVED** [review]

**Pull request merged**